week2/hw2/memberSerch.js

```diff
@@ -0,0 +1,181 @@
+});
+
+delBtn.addEventListener("click", () => {
+  const checked = document.querySelectorAll(".check:checked");
+  const checkedTrs = Array.from(checked).map((checkedTr) => parseInt(
+    checkedTr.closest("tr").id
+  ))
+  const newMember = getData.filter(member => {
+    return !checkedTrs.includes(member.id);
+});
+  console.log(newMember);
+
+  localStorage.setItem("membersData", JSON.stringify(newMember))
+
+  // window.location.reload();
+  tbody.innerHTML = "";
+  addTable(newMember);
+});
+
+
+
+selectAll.addEventListener("change", () => {
+  const isChecked = selectAll.checked;
+
+  const dataCheckbox = tbody.querySelectorAll(".check")
+  dataCheckbox.forEach(checkbox => {
+    checkbox.checked = isChecked;
+  });
+});
+
+showBtn.addEventListener("click", () => {
+  modal.showModal();
+});
+
+closeModalBtn.addEventListener("click", () => {
+  modal.close();
+});
+
+addDataBtn.addEventListener("click", () => {
+  const name = document.getElementById("name").value;
+  const englishName = document.getElementById("englishName").value;
+  const github = document.getElementById("github").value;
+  const gender = document.getElementById("gender").value;
+  const role = document.getElementById("role").value;
+  const firstWeekGroup = document.getElementById("firstWeekGroup").value;
+  const secondWeekGroup = document.getElementById("secondWeekGroup").value;
+
+  if (!name || !englishName || !github || !gender || !role || !firstWeekGroup || !secondWeekGroup) {
+    alert("모든 필드를 입력해주세요.");
+    return;
+  }
+
+  const newMember = {
+    id: getData.length + 1,
+    name,
+    englishName,
+    github,
+    gender,
+    role,
+    firstWeekGroup,
+    secondWeekGroup
+  };
+
+  console.log(newMember);
+
+  getData.push(newMember);
+  console.log(getData);
+
+  localStorage.setItem("membersData", JSON.stringify(getData));
+  
+  const row = document.createElement("tr");
+
+  row.innerHTML = `
+    <td><input type="checkbox" class="check"></td>
+    <td>${newMember.name}</td>
+    <td>${newMember.englishName}</td>
+    <td>${newMember.github}</td>
+    <td>${newMember.gender}</td>
+    <td>${newMember.role}</td>
+    <td>${newMember.firstWeekGroup}</td>
+    <td>${newMember.secondWeekGroup}</td>
+  `;
+
+  tbody.appendChild(row);
+
+  document.getElementById("name").value = '';
+  document.getElementById("englishName").value = '';
+  document.getElementById("github").value = '';
+  document.getElementById("gender").value = '';
+  document.getElementById("role").value = '';
+  document.getElementById("firstWeekGroup").value = '';
+  document.getElementById("secondWeekGroup").value = '';
+
+  modal.close();
+
+  window.location.reload();
+});
+
+modal.addEventListener('click', function(event) {
+  const target = event.target;
+  const rect = target.getBoundingClientRect();
+  if (rect.left > event.clientX || rect.right < event.clientX ||
+    rect.top > event.clientY || rect.bottom < event.clientY) {
+      modal.close();
+    }
+})
```

Comment:
백드롭을 구현하기 위해 getBoundingCLientRect()라는 매서드를 사용하셨네요. 그리고 rect및 event 객체의 x,y 좌표값을 그때 그때 계산해서 모달 바깥 영역인지 체크하는 코드같은데 

```suggestion
modal.addEventListener("click", (e) => {
  if (e.target === e.currentTarget) modal.close();
});
```
이렇게 하면 훨씬 쉽게 구현할 수 있답니다.
e.target은 이벤트객체의 현재 클릭된 요소, 즉 사용자가 실제로 현재 클릭한 부분이 선택됩니다. 반면 e.currentTarget은 이벤트 리스너가 등록된 요소인 modal이 선택됩니다.
따라서 모달 내부 영역을 클릭하면 해당 부분의 div, input 등등이 선택되고, 모달 외부 영역을 클릭하면 dialog 태그에 의해 바깥영역이 dialog태그에 해당되어, dialog태그가 선택되면서 modal.close()가 실행됩니다.

Reply:
이 부분..! 저도 화랑님의 코리를 보며 많이 배워갑니다!!